libs/accounts/src/lib/accounts-data-provider.ts

```diff
@@ -67,3 +71,82 @@ export const accountsDataProvider = makeDataProvider<
   getData,
   getDelta,
 });
+
+export interface AccountFields extends AccountFieldsFragment {
+  available: string;
+  used: string;
+  deposited: string;
+  balance: string;
+  breakdown?: AccountFields[];
+}
+
+const USE_ACCOUNT_TYPES = [
+  AccountType.ACCOUNT_TYPE_MARGIN,
+  AccountType.ACCOUNT_TYPE_BOND,
+  AccountType.ACCOUNT_TYPE_FEES_INFRASTRUCTURE,
+  AccountType.ACCOUNT_TYPE_FEES_LIQUIDITY,
+  AccountType.ACCOUNT_TYPE_FEES_MAKER,
+  AccountType.ACCOUNT_TYPE_PENDING_TRANSFERS,
+];
+
+const getAssetIds = (data: AccountFieldsFragment[]) =>
+  Array.from(new Set(data.map((a) => a.asset.id))).sort();
+
+const getTotalBalance = (accounts: AccountFieldsFragment[]) =>
+  accounts.reduce((acc, a) => acc + BigInt(a.balance), BigInt(0));
+
+export const getAccountData = (
+  data: AccountFieldsFragment[]
+): AccountFields[] => {
+  return getAssetIds(data).map((assetId) => {
+    const accounts = data.filter((a) => a.asset.id === assetId);
+    return accounts && getAssetAccountAggregation(accounts, assetId);
+  });
+};
+
+const getAssetAccountAggregation = (
+  accountList: AccountFieldsFragment[],
+  assetId: string
+): AccountFields => {
+  const accounts = accountList.filter((a) => a.asset.id === assetId);
+  const available = getTotalBalance(
+    accounts.filter((a) => a.type === AccountType.ACCOUNT_TYPE_GENERAL)
+  );
+
+  const used = getTotalBalance(
+    accounts.filter((a) => USE_ACCOUNT_TYPES.includes(a.type))
+  );
+
+  const balanceAccount: AccountFields = {
+    asset: accounts[0].asset,
+    balance: available.toString(),
+    type: AccountType.ACCOUNT_TYPE_GENERAL,
+    available: available.toString(),
+    used: used.toString(),
+    deposited: (available + used).toString(),
+  };
+
+  const breakdown = accounts
+    .filter((a) => AccountType.ACCOUNT_TYPE_GENERAL !== a.type)
```

Comment:
suggestion (non blocking): I think this code would be more readable if you filter'd in the USE_ACCOUNT_TYPES and then just mapped over them. Then I don't think you need the addtional ternary checks within the map callback itself

Reply:
The other day I received a comment that `AccountType.ACCOUNT_TYPE_GENERAL !== a.type` would make it more readable. Plus, I want to display as much as I can in the breakdown, almost every account entry except what is in general.